src/client/calculate-updates.js

```diff
@@ -0,0 +1,82 @@
+import { find } from 'lodash';
+import { HOUR } from '../utils/constants';
+
+export const DEFAULT_NEXT_UPDATE = 1 * HOUR;
+
+export default function calculateUpdates(
+	requirementsByEndpoint,
+	endpointsState,
+	now = new Date()
+) {
+	const updateInfo = { updates: [], nextUpdate: DEFAULT_NEXT_UPDATE };
+	appendUpdatesForEndpoints( updateInfo, [], requirementsByEndpoint, endpointsState, now );
+	return updateInfo;
+}
+
+function appendUpdatesForEndpoints( updateInfo, endpointPath, requirementsByEndpoint, endpointsState, now ) {
```

Comment:
Do we want docblocks on these methods?

Reply:
Yeah, I suppose. They're not public, but some extra explanation can't hurt. I'll update and commit. Thanks!